modules/twinkleblock.js

```diff
@@ -1103,7 +1222,12 @@ Twinkle.block.callback.preview = function twinkleblockcallbackPreview(form) {
 		hardblock: Twinkle.block.isRegistered ? form.autoblock.checked : form.hardblock.checked,
 		indefinite: (/indef|infinit|never|\*|max/).test(form.template_expiry ? form.template_expiry.value : form.expiry.value),
 		reason: form.block_reason.value,
-		template: form.template.value
+		template: form.template.value,
+		partial: form.partial ? form.partial.value : false,
+		pagerestrictions: $(form.pagerestrictions).val() || [],
```

Comment:
Yup 👍 — held over from while I was figuring out 1259-1260.

Reply:
Actually, no, I'd forgotten why I put this in: if the user disables blocks before enabling the partial status, there's no `form.pagerestrictions`, so it returns `undefined` rather than `[]`.  It's either this or turn:

`params.pagerestrictions.length || params.namespacerestrictions.length)`

into

`((params.pagerestrictions && params.pagerestrictions.length) || (params.namespacerestrictions && params.namespacerestrictions.length))`

This seemed easier.